File: hw/codena/u3/hw24CompositePatterCode/Javascript/CompositePattern/model/President.js
import Supervisor from "./Supervisor.js";

export default class President extends Supervisor {

    static instance = null;

    constructor(name) {

        if (President.instance) {
            President.instance.name = name;
            return President.instance;
        }

        super();

        this.title = "President";
        this.name = name;

        President.instance = this;
    }

}

File: hw/codena/u3/hw24CompositePatterCode/Python/composite_pattern/model/president.py
from model.supervisor import Supervisor


class President(Supervisor):

    __instance = None

    def __new__(cls, name):
        if cls.__instance is None:
            cls.__instance = super().__new__(cls)
        return cls.__instance

    def __init__(self, name):
        if not hasattr(self, "_initialized"):
            super().__init__()
            self.title = "President"
            self._initialized = True

        self.name = name